charts/terra-argocd-app/templates/cluster.yaml

```diff
@@ -0,0 +1,45 @@
+{{- if .Values.releaseType | eq "cluster" }}
+# Generates an ArgoCD application for a Terra cluster release
+{{- $release := required "releaseName is required" .Values.releaseName }}
+{{- $cluster := required "targetName is required" .Values.targetName }}
+{{- if not ( .Values.targetName | eq .Values.clusterName ) }}
```

Comment:
`{{ if not (eq .Values.targetName .Values.clusterName) }}` I believe is what you're looking for here...? 
since $cluster is required, can we reference it here?

Reply:
`eq .Values.targetName .Values.clusterName` is the same as `.Values.targetName | eq .Values.clusterName`, just a different style. I find the interleaved syntax more legible because of its similarity to other programming languages (`if a == b`) 🤷

Reply:
I see. Here it was a little confusing because the output before the pipe is an input for after the pipe. I didn't know if you were trying to evaluate something prior to checking if the values are equal.